yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -290,8 +290,18 @@ private[yarn] class YarnAllocator(
       location: String,
       containersToUse: ArrayBuffer[Container],
       remaining: ArrayBuffer[Container]): Unit = {
+    // SPARK-6050: certain Yarn configurations return a resource configuration that doesn't match
+    // the request; for example, capacity scheduler + DefaultResourceCalculator. Allow users in
+    // those situations to disable resource matching.
+    val matchingResource =
+        if (sparkConf.getBoolean("spark.yarn.container.matchAnyResource", false)) {
```

Comment:
Also as @mridulm mentioned we are basically just matching what we got back which disables all checks. Before we were checking that memory was atleast big enough.

  private def isResourceConstraintSatisfied(container: Container): Boolean = {
    container.getResource.getMemory >= (executorMemory + memoryOverhead)
  }